### PR TITLE
Add dockerfiles for front and back

### PR DESCRIPTION
**Pull request by zaymat**

### CODE REVIEW

File: cmd/api/main.go [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
 func main() {
-	d, err := db.NewDGraph()
+	// Get filename from args
+	filename := "../../config.json"
+	if len(os.Args) == 2 {
+		filename = os.Args[1]
+	}
+
+	// Get config from file
+	file, err := os.Open(filename)
```
Comment (anatolebeuzon) [inline cmd/api/main.go]: nit: we can be a slightly more straightforward by using `ioutil.ReadFile` and `json.Unmarshal`

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1 @@
+frontend/node_modules
```
Comment (anatolebeuzon) [inline .dockerignore]: nit: no newline at end of file :-) https://unix.stackexchange.com/a/18789/250345

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,6 @@
 
 # Go binaries
 cmd/crawler/crawler
+
+# Config file
+config.json
```
Comment (anatolebeuzon) [inline .gitignore]: same (newline)

File: cmd/crawler/main.go [hunk 4]
```diff
@@ -19,8 +20,8 @@ func main() {
 
 	log.Println(args)
 
-	if len(args) != 3 {
-		log.Println("Usage: crawler <prefix> <startTitle> <nWorkers>")
+	if len(args) != 3 || len(args) != 4 {
```
Comment (anatolebeuzon) [inline cmd/crawler/main.go]: I suppose you meant `&&` instead of `||`

File: cmd/crawler/main.go [hunk 5]
```diff
@@ -31,7 +32,26 @@ func main() {
...
 
-	client, _ := db.NewDGraph()
+	// Get filename from args
+	filename := "../../config.json"
+	if len(args) == 4 {
+		filename = args[3]
+	}
+
+	// Get config from file
+	file, err := os.Open(filename)
```
Comment (anatolebeuzon) [inline cmd/crawler/main.go]: same nit: could use `ioutil.ReadFile` and `json.Unmarshal` instead

File: cmd/crawler/main.go [hunk 6]
```diff
@@ -31,7 +32,26 @@ func main() {
...
+	}
+
+	decoder := json.NewDecoder(file)
+	var config db.Config
+	err = decoder.Decode(&config)
+	if err != nil {
+		log.Fatal("Couldn't parse config file")
+	}
+
+	client, _ := db.NewDGraph(config)
```
Comment (anatolebeuzon) [inline cmd/crawler/main.go]: I know it's not directly related to this PR, but can we catch the error that can be returned by `NewDGraph` here? 😄

File: docker/api.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+ENV GOPATH=/go
+
+RUN mkdir -p /go/src/api
+COPY ./cmd/api /go/src/api
+COPY config.json.template /go/src/api/config.json
+
+WORKDIR /go/src/api
+RUN go get ./...
+RUN go build . 
+CMD ["./api", "./config.json"]
```
Comment (anatolebeuzon) [inline docker/api.Dockerfile]: no newline x)

File: docker/front.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
+FROM node:13
+
+RUN mkdir /front
+COPY ./frontend /front
+
+WORKDIR /front
+
+RUN yarn install
+
+CMD ["yarn", "serve"]
```
Comment (anatolebeuzon) [inline docker/front.Dockerfile]: no newline x)

**Comment by zaymat** [comment]
Fixed

**Pull request merged**